### DISCUSSION TITLE
allow editing an existing remote's URL

### DISCUSSION THREAD

**Discussion by aspiers** [magit/magit#4652] [Support] [answered]
## The problem

As we all know, to edit a remote's URL, we can press `M` then `u` (or `M` then `C`, select the remote, then `u`).

However, this offers a blank prompt, rather than an editable version of the current URL.  This means that if I want to change it to something similar (and let's face it, many remote URLs have a *lot* in common), I have to type it all over again, or copy from elsewhere and then yank.

## Proposed solution

It would be much nicer if by default it populated the prompt for the new URL with the existing one, probably leaving the point at the end of the URL, since it's more likely that something towards the end of the URL will need to be changed than something near the beginning.

If the user really wants to start typing from scratch, they can then easily type `C-a C-k` first.  But in the majority of cases it will be preferable to edit rather than start from scratch.

**Comment by tarsius** [comment, marked as answer]
Or you could press `M-p` to get the previous value from the completion history (where `transient` explicitly puts it).

**Reply by aspiers** [reply]
Ah, thanks!  I did actually try `M-p` a few hours ago before filing this discussion, and I could have sworn it didn't work as I (then) expected, but it seems to be working fine now.  Did that feature appear in the last few months, by any chance?  I just upgraded `magit` and `transient` for the first time in many months - but maybe that was shortly after filing this discussion rather than before.  Although I experienced a bunch of weird issues during the upgrade[^1] so can't remember the exact order everything happened.

I do remember that after typing the URL manually, it failed to set the remote for some unrelated reason, and then I tried to retrieve the manually typed URL via `M-p` but it failed so I had to type it.  Maybe the problem was that I was expecting the last typed value to be the first value retrieved via `M-p`, rather than the remote's current URL, and that confused me sufficiently to assume that `M-p` didn't work at all.

[^1]: Like `transient--emergency-exit((invalid-function transient--with-suspended-override)`, which I think was probably cured by removing some potentially old `.elc` files and then restarting emacs, even though I had already tried restarting several times, and had explicitly asked `straight.el` to rebuild all the packages ...

**Reply by tarsius** [reply]
I don't remember when I added that, so probably quite a while ago.
